lib/manageiq/appliance_console/postgres_admin.rb

```diff
@@ -320,6 +331,83 @@ class << self
     ensure
       File.delete(error_path) if File.exist?(error_path)
     end
+
+    private_class_method def self.prepare_restore(backup_type, dbname)
+      if application_connections?
+        message = "Database restore failed. Shut down all evmserverd processes before attempting a database restore"
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+
+      disable_replication(dbname)
+
+      if connection_count(backup_type) > 1
+        message = "Database restore failed. #{connection_count - 1} connections remain to the database."
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+    end
+
+    private_class_method def self.application_connections?
+      result = [{"count" => 0}]
+
+      with_pg_connection do |conn|
+        result = conn.exec("SELECT COUNT(pid) FROM pg_stat_activity WHERE application_name LIKE '%MIQ%'")
+      end
+
+      result[0]["count"] > 0
+    end
+
+    private_class_method def self.disable_replication(dbname)
```

Comment:
I think this will handle all of the necessary concerns that were brought up previously:

https://github.com/ManageIQ/manageiq-appliance_console/pull/160/files/650c62b4dfc42ba519df50c8a8553b5aee848060#diff-4f5a9d143c19dfcadcdb0467ca432e5c0eba904c896862e8af14277978ec1479

I didn't delete the `MiqRegion`, as is currently done in `PglogicalSubscription`, but since this is a restore, that data will be blown away anyway, so we only need the parts that will be used prevent an error due to the HA code preventing a restore (which I think this is all of it... but I will double check).

I condensed a lot of what was in `PglogicalSubscription` to be a single method, so much of what is used here is spread out in the class between `.find`, `.find_all`, `.delete`, `.safe_delete`, and others.  I also removed some of the column id conversion as I didn't think it was necessary, and just referenced it as `sub_id` for the purposes of the `.drop_subscription` portion of the code below.

Reply:
So I did test this on a local appliance using:

https://github.com/NickLaMuro/manageiq-db_backup-tests/pull/1

And it worked just fine, so at least from a non-HA appliance setup, these new changes don't seem to break anything.  Don't really know how to set up a HA appliance to test the scenarios that were present in https://bugzilla.redhat.com/show_bug.cgi?id=1444993 so that is still a unknown, but the base case at least still works.